src/modules/organisations/organisations.service.ts

```diff
@@ -169,6 +171,43 @@ export class OrganisationsService {
     }
   }
 
+  async searchOrganisationMember(orgId: string, searchTerm: string, searchMemberQueryDto: SearchMemberQueryDto) {
+    const organisationMembers = await this.organisationMemberRepository.find({
```

Comment:
All your logic should have ended here. There should only be one organisation member with a particular user and organisation id. Everything else after this should be you formating and returning your response. I think the rest of your code would be unnecessary if you simply optimize your query appropriately and use findOne instead

Reply:
The purpose is to search organisation members not an organisation member